infra/main.bicep

```diff
@@ -0,0 +1,314 @@
+param springBootContainerAppName string = ''
+
+@description('Set if the SpringBoot Telemetry Container aAp already exists.')
+param springBootAppExists bool = false
+
+/******************************************************************************/
+/*                                 VARIABLES                                  */
+/******************************************************************************/
+
+var abbreviations = loadJsonContent('./abbreviations.json')
+
+// tags that should be applied to all resources.
+var tags = {
+  // Tag all resources with the environment name.
+  'azd-env-name': environmentName
+}
+
+/******************************* Resource Names *******************************/
+
+
+// Generate a unique token to be used in naming resources.
+// Remove linter suppression after using.
+#disable-next-line no-unused-vars
+var resourceToken = toLower(uniqueString(subscription().id, environmentName, location))
+
+@description('Name of the environment with only alphanumeric characters. Used for resource names that require alphanumeric characters only.')
+var alphaNumericEnvironmentName = replace(replace(environmentName, '-', ''), ' ', '')
+
+var _containerAppsEnvironmentName = !empty(containerAppsEnvironmentName) ? containerAppsEnvironmentName : take('${abbreviations.appManagedEnvironments}${environmentName}', 60)
+var _containerRegistryName = !empty(containerRegistryName) ? containerRegistryName : take('${abbreviations.containerRegistryRegistries}${take(alphaNumericEnvironmentName, 35)}${resourceToken}', 50)
+var _logAnalyticsName = !empty(logAnalyticsWorkspaceName) ? logAnalyticsWorkspaceName : take('${abbreviations.operationalInsightsWorkspaces}${environmentName}', 63)
+var _applicationInsightsName = !empty(applicationInsightsName) ? applicationInsightsName : take('${abbreviations.insightsComponents}${environmentName}', 255)
+var _applicationInsightsDashboardName = !empty(applicationInsightsDashboardName) ? applicationInsightsDashboardName : take('${abbreviations.portalDashboards}${environmentName}', 160)
+var _quarkusContainerAppName = !empty(quarkusContainerAppName) ? quarkusContainerAppName : take('${abbreviations.appContainerApps}quarkus-${environmentName}', 32)
+var _springBootContainerAppName = !empty(springBootContainerAppName) ? springBootContainerAppName : take('${abbreviations.appContainerApps}spring-boot-${environmentName}', 32)
+var _postgresFlexibleServerName = !empty(postgresFlexibleServerName) ? postgresFlexibleServerName : take(toLower('${abbreviations.dBforPostgreSQLServers}${take(environmentName, 44)}-${resourceToken}'), 63)
+var _keyVaultName = !empty(keyVaultName) ? keyVaultName : take('${abbreviations.keyVaultVaults}${take(alphaNumericEnvironmentName, 8)}${resourceToken}', 24)
+var _keyVaultSecrets = [
+  {
+    name: 'postgres-admin-password'
+    value: postgresAdminPassword
```

Comment:
We are trying to move to passwordless auth for PG but that is a larger change (as it also effects code)

Reply:
Yes, it wouldn't be a great fit for this example as it is already very complex. Or WDYT @brunobat and @jeanbisutti ?

Reply:
I would propose we make this a separate issue. I'm not sure the exact impact it would have on the native part,